compiler/vm.nim

```diff
@@ -144,7 +144,19 @@ proc derefPtrToReg(address: BiggestInt, typ: PType, r: var TFullReg, isAssign: b
   of tyFloat: fun(floatVal, float, rkFloat)
   of tyFloat32: fun(floatVal, float32, rkFloat)
   of tyFloat64: fun(floatVal, float64, rkFloat)
-  else: return false
+  of tyObject:
+    if isAssign:
+      let lhs = cast[PNode](address)
+      let rhs = r.node
+      doAssert lhs.len == rhs.len, $(lhs.len, rhs.len)
```

Comment:
"assertion error (3, 4)" is a bad error message.

Reply:
changed to `internalError(c.config, c.debug[pc], "derefPtrToReg failed")` which shows the line info